### ISSUE TITLE
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0'

### ISSUE THREAD

**Issue by bill-hudacek** [qarmin/czkawka#463]
I have a 4 TB external drive. I save backups to subdirs on this drive, dated, so there are (e.g.,) these subdirs:

dr-xr-xr-x 16 root root      4096 2020-12-18 13:37 2021-01-03/
dr-xr-xr-x 16 root root      4096 2020-12-18 13:37 2021-02-16/
dr-xr-xr-x 15 root root      4096 2021-05-05 19:40 2021-05-05/
dr-xr-xr-x 15 root root      4096 2021-08-11 11:13 2021-08-10/
dr-xr-xr-x 15 root root      4096 2021-05-05 19:40 2021-08-30/
dr-xr-xr-x 15 root root      4096 2021-09-22 22:02 2021-09-27/

Space getting a bit tight. fslint-gui appears to be orphanware, so I got czkawka_gui installed. Looking for duplicate files, it ran for /days/ (drive is USB-3 connected RAID-0, a Seagate 'Fast HDD' model SRD00M1).

This morning, I saved the list (there's over 1 TB of dupe files - nice!). I'm glad I saved the list as the moment I clicked 'hard-link' button, it crashed. This is the only output:

thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', czkawka_gui/src/connect_button_hardlink.rs:109:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

I don't care to run it for days again - but I'd be happy to try to reproduce with a smaller subset of the above dir list. Any tips aside from RUST_BACKTRACE? Any debugging steps just before I click that hard-link button?

At least I had a list of the most monstrous files - got back many GB just processing those myself in a shell script....

I did not get a screen capture, sorry to say, just before clicking that button.....but there are 21,522,467 used in the XFS filesystem.

**Comment by qarmin** [comment]
The problem happens because, when clicking at hardlink or symlink button there is no selected any record
I tested various configurations, but never tried to not select anything ¯\\\_(ツ)\_/¯
This will be fixed by https://github.com/qarmin/czkawka/pull/462, and maybe backported to 3.3.2 with https://github.com/qarmin/czkawka/pull/464

Fixed binaries can be found here - https://github.com/qarmin/czkawka/actions?query=branch%3A3.3.2

**Comment by bill-hudacek** [comment]
Thanks for that quick reply!

I had selected all - before I saved the output. Perhaps it de-selected all after (but I don't remember if the display updated...or, indeed, what was showing when I clicked the button!). Sorry about that. You try to catch every thing that you are doing.....I just missed this one.

I can test with perhaps 1 million files, and try it with nothing selected, or everything selected - if you like. 

For now I'll get the updated version. Time for a dumb question. I can't find how to switch to your unstable version using flatpak. Should I uninstall the flatpak stable version and use another approach (git clone)?

(I'm running Czkawka 3.3.1 right now, probably should have mentioned that too...)

Thanks again.

**Comment by qarmin** [comment]
I only provide stable versions for flatpak(I didn't know that exists something like unstable builds)
For me the simplest way to get installed the latest build with fix is to run this command(require at least Rust 1.53.0 installed)
```
cargo install --git https://github.com/qarmin/czkawka  czkawka_gui --branch 3.3.2
```

**Comment by bill-hudacek** [comment]
OK, I'll try using 'cargo'. Thanks.

**Comment by bill-hudacek** [comment]
Hmmmm. After cargo command you gave, version info in the GUI is still Czkawka 3.3.1 - just mislabeled?

Strings run on the binary gave this bit of info:

<object class="GtkAboutDialog" id="about_dialog">
      <property name="can-focus">False</property>
      <property name="window-position">center</property>
      <property name="type-hint">dialog</property>
      <property name="program-name">Czkawka</property>
*     <property name="version">3.3.1</property>
      <property name="comments" translatable="yes">2020 - 2021  Rafa
   Mikrut(qarmin)
  This program is free to use and will always be.
  
  It's just a tag in the metadata. I'm not going to worry about it. I'll run some tests here, with increasingly large numbers of files, let's see what I find. I hope & expect that this particular NPE/refcount-error is gone.

**Comment by qarmin** [comment]
This is only PR is built on top 3.3.1 version, only 4 lines were added.
I don't change for now version number, because I'm not sure in which exactly version such commit will be included(probably 3.3.2, but it is also possible that 3.4.0)
